install.sh

```diff
@@ -994,11 +994,27 @@ case "${SHELL}" in
     ;;
 esac
 
-cat <<EOS
+in_shell_profile=$(grep -qs "eval \"\$(${HOMEBREW_PREFIX}/bin/brew shellenv)\"" "${shell_profile}")
+
+# FIXME: figure out why this give a "command not found" error
```

Comment:
Which command is not found?

Reply:
The error occurs when using the `$in_shell_profile` variable in the if statements.

Reply:
That's because `grep -qs` always returns nothing.

Reply:
Assigning to a variable and checking its value is in general less robust than directly checking for the condition (`if grep`) which returns *the exit code*, the true marker of a command’s success/failure.

With @MikeMcQuaid’s suggestion below of a nested `if`, the command can again be used directly instead of assigned to a variable.